src/pages/iv-categories/forms/CategorySiteForm.js

```diff
@@ -0,0 +1,161 @@
+import { useFormik } from 'formik'
+import { DataGrid } from 'src/components/Shared/DataGrid'
+import FormShell from 'src/components/Shared/FormShell'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { useContext, useEffect } from 'react'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import toast from 'react-hot-toast'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { ControlContext } from 'src/providers/ControlContext'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import { Grid } from '@mui/material'
+import { useForm } from 'src/hooks/form'
+
+const CategorySiteForm = ({ store, labels, maxAccess }) => {
+  const { recordId } = store
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const { formik } = useForm({
+    maxAccess,
+    enableReinitialize: true,
+    validateOnChange: true,
+
+    initialValues: {
+      sites: [
+        {
+          id: 1,
+          categoryId: recordId,
+          siteId: '',
+          name: '',
+          reference: '',
+          isLocked: false,
+          siteId: ''
+        }
+      ]
+    },
+    onSubmit: values => {
+      postdata(values)
+    }
+  })
+
+  const postdata = obj => {
+    const sites = obj?.sites?.map(({ categoryId, ...rest }) => ({
+      categoryId: recordId,
+
+      ...rest
+    }))
+
+    const list = {
+      categoryId: recordId,
+      sites: sites
+    }
+    postRequest({
+      extension: InventoryRepository.CategorySites.set2,
+      record: JSON.stringify(list)
+    })
+      .then(res => {
+        toast.success(platformLabels.Edited)
+        getData()
+      })
+      .catch(error => {})
+  }
+
+  const columns = [
+    {
+      component: 'textfield',
+      name: 'reference',
+      label: labels.reference,
+      props: {
+        readOnly: true
+      }
+    },
+    {
+      component: 'textfield',
+      name: 'name',
+      label: labels.name,
+      props: {
+        readOnly: true
+      }
+    },
+    {
+      component: 'checkbox',
+      label: labels.isLocked,
+      name: 'isLocked'
+    }
+  ]
+  function getData() {
+    getRequest({
```

Comment:
refactor this function by removing several catches blocks, this can be done by adding async/await for each api call